src/main/resources/bamm/characteristic/2.0.0/characteristic-definitions.ttl

```diff
@@ -190,6 +190,7 @@ bamm-c:TimeSeries rdfs:subClassOf bamm-c:SortedSet ;
    bamm:dataType bamm-e:TimeSeriesEntity .
 
 bamm-c:Enumeration rdfs:subClassOf bamm:Characteristic ;
+   rdfs:subClassOf mmm:NamedConcept ;
```

Comment:
Once the `bamm:name` attribute was removed, the synthetic name generated for anonymous nodes was used for the name of the corresponding Java Enum in the code generation. This lead to generated Java Enums being called e.g. `Enumeration25347356`. IMHO this leads to a bad developer experience. Hence I changed the Enumeration to be a named element.

Reply:
I get the point, but I'm not sure making this one arbitrary model element an exception "just because" it happens to have its name used in the code generation is a good idea. In a different code generation approach or code generation for a different language this might not be required; and in the other way round, in a different code generation approach, other Characteristics could use a name too. Although I don't want to be dogmatic about this, this strikes me as (a little too?) pragmatist.
I'd prefer something like the following:
1. Add the recommendation to name all Characteristics, unless there is a good reason not to. But no Characteristic is technically a mmm:NamedConcept.
2. If the Enumeration is named, the name is used as an identifier in the code.
3. If it isn't, from a modeling/semantic point of view, its name does not convey any meaning. This means that in the code it could be named "anything".
4. To get readable code for unnamed Characteristics, improve the "synthetic identifier" algorithm (maybe only for Enumerations, maybe for other things as well): Incorporate context such as the Characteristic's preferredName if present, and in particular, the Property it's used in. If the Characteristic is an anoymous node, this means it is used only in exactly one Property; so the Property's name can help identify the Characteristic.
What do you think?

Reply:
Agreed, the approach is too Java centric.

With regards to extending the "synthetic identifier" algorithm, I would include the Property's name and ignore the Characteristic's preferred name(s). Then we do not have to handle issues such as having to choose from multiple preferred names.

Reply:
Sounds reasonable.